File: Modules/SMES/MES/SIE.Web.MES/Outsourcing/Commands/OutboundRecordAddCommand.js
SIE.defineCommand("SIE.Web.MES.Outsourcing.Commands.OutboundRecordAddCommand", {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加出库记录", group: "edit", iconCls: "icon-AddEntity icon-green" },
    canExecute: function (view) {
        const outsourcingRequest = view._parent.getCurrent();
        if (outsourcingRequest == null) {
            return false;
        }
        if (view._parent != null && view._parent.getCurrent() != null && view._parent.getCurrent().getData().OutsourcingState == 20) {
            return false;
        }
        return !(outsourcingRequest.isNew());
    },

    onItemCreated: function (entity) {
        var model = entity.model;
        var me = this;
        this.view.execute({
            data: model,
            success: function (res) {
                var data = res.Result;
                if (data) {
                    entity.setState(data.State);
                }

            }
        }, me.view);
    },
});

File: Modules/SMES/MES/SIE.Web.MES/Outsourcing/Commands/OutboundRecordAddCommand.cs
using SIE.MES.Outsourcing;
using SIE.Web.Command;
using System;

namespace SIE.Web.MES.Outsourcing.Commands
{
    /// <summary>
    /// 添加出库记录
    /// </summary>
    public class OutboundRecordAddCommand : ViewCommand
    {
        /// <summary>
        /// 添加出库记录
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        /// <exception cref="NotImplementedException"></exception>
        protected override object Excute(ViewArgs args, string scope)
        {
            var processingOutbound = args.Data.ToJsonObject<ProcessingOutbound>();
            processingOutbound.SourceId = 0;
            processingOutbound.State = OutsourcingDetailState.Created;
            return processingOutbound;
        }
    }
}
